lib/classes/ConfigSchemaHandler/index.js

```diff
@@ -216,6 +216,28 @@ class ConfigSchemaHandler {
     });
   }
 
+  extendFunctionEventDefinition(providerName, name, configSchema) {
+    if (this.serverless.service.provider.name !== providerName) {
+      return;
+    }
+
+    const existingEventDefinitionIndex = this.schema.properties.functions.patternProperties[
+      FUNCTION_NAME_PATTERN
+    ].properties.events.items.anyOf.findIndex(
```

Comment:
It might be cleaner to use `find` and save result to e.g. `eventDefinition` property. Then we don't have to repeat this long access path below